tests/runner.py

```diff
@@ -484,20 +484,30 @@ def tearDown(self):
       #     assert not temp_file.endswith('.ll'), temp_file
       #     # TODO assert not temp_file.startswith('emscripten_'), temp_file
 
+  def _check_settings_key(self, key):
+    self.assertIn(key, shared.Settings)
+
   def get_setting(self, key):
+    self._check_settings_key(key)
     if key in self.settings_mods:
       return self.settings_mods[key]
-    return Settings[key]
+    return shared.Settings[key]
 
   def set_setting(self, key, value=1):
+    self._check_settings_key(key)
     if value is None:
       self.clear_setting(key)
+    if not self.has_changed_setting(key):
+      # Sanity check: tests shouldn't be setting value to thier defaults
+      assert value != shared.Settings[key], 'attempt to set_setting to default value'
```

Comment:
Use `self.assertNotEqual`?

Reply:
Good question.   I think for me its because this an assertion about the test itself, not a condition the test is checking.  So it will result in a test error rather than a test failure.   I think this makes sense.  

Another way of putting it, if this assert is hit is means the test has a logical error, not that the system under test failed.